### ISSUE TITLE
Chunked dispatching breaks multiple matching subscriptions

### ISSUE THREAD

**Issue by kuema** [crossbario/crossbar#1038]
After upgrading from Crossbar `16.10.1` to `17.3` I noticed that published messages do no longer
get properly dispatched across all matching clients (different instances, even on different hosts) when using pattern-based subscriptions. Only the client with the "closest match" would receive publication messages.

### Version information

```
 Crossbar.io        : 16.10.1
   Autobahn         : 0.16.1 (with JSON, MessagePack, CBOR, UBJSON)
   Twisted          : 16.5.0-EPollReactor
   LMDB             : 0.92/lmdb-0.9.18
   Python           : 2.7.9/CPython
 OS                 : Linux-3.16.0-4-amd64-x86_64-with-debian-8.7
 Machine            : x86_64
 Release key        : RWQ2MDk26PKBMNUZG2Jok1tMBB1SKyci+N7dtcep8jrikTl4NvI1Rnux
```

```
 Crossbar.io        : 17.3.1 (Crossbar.io COMMUNITY)
   Autobahn         : 0.18.1 (with JSON, MessagePack, CBOR, UBJSON)
   Twisted          : 17.1.0-EPollReactor
   LMDB             : 0.92/lmdb-0.9.18
   Python           : 2.7.9/CPython
 OS                 : Linux-3.16.0-4-amd64-x86_64-with-debian-8.7
 Machine            : x86_64
 Release key        : RWRGxs0uB6vQemWGPeBt7nLG53ADoOtZy/Vve3BjlaXDqkCiVhCs7Zrv

```

### Test scenario

(I used a simple NodeJS/Autobahn 0.12 client for testing)

- Client A: Publish messages to `com.myapp.topic1`
- Client B: Subscribe to `com..topic1` with `wildcard` matching
- Client C: Subscribe to `com.myapp` with `prefix` matching
- Client D: Subscribe to `com` with `prefix` matching
- Client E: Subscribe to `com.myapp.topic1` with `exact` matching

Start all client instances in this order.

### Behavior
With Crossbar 16.10.1 *all* subscriber clients (B to E) would receive messages simultaneously. **This is the expected behavior.**

With Crossbar 17.3, after starting client C each previous client stops receiving messages. Closing the clients in reverse order, the previous client receives again.

It seems to me, that at some point between the two releases the match policy hierarchy as described for
[registrations](https://github.com/crossbario/crossbar/blob/master/docs/pages/programming-guide/rpc/Pattern-Based-Registrations.md#conflict-resolution) (where it is desired) got implemented for subscriptions too.

**Comment by kuema** [comment]
I was able too narrow down the problem a bit further.

I just tried Crossbar `17.2.1` with the included Autobahn `0.17.2` and everything is still working fine - all clients receive publications simultaneously.

```
 Crossbar.io        : 17.2.1 (Crossbar.io COMMUNITY)
   Autobahn         : 0.17.2 (with JSON, MessagePack, CBOR, UBJSON)
   Twisted          : 17.1.0-EPollReactor
   LMDB             : 0.92/lmdb-0.9.18
   Python           : 2.7.9/CPython
 OS                 : Linux-3.16.0-4-amd64-x86_64-with-debian-8.7
 Machine            : x86_64
 Release key        : RWTmo7YQ1mmpesRyo4DYay3REDoJ7HsafXYcfhK3O+lM9JbeAAGSXUTh
```

It seems the new publishing behavior got introduced with the transition to Autobahn `0.18.1` starting with Crossbar `17.3.1`. I hope this information is helpful in tracking down the problem more quickly.

**Comment by oberstet** [comment]
I believe the issue should be fixed in above commit, and covered in below new integration test:

* https://github.com/crossbario/autobahn-js/blob/master/test/test_pubsub_multiple_matching_subs.js
* https://github.com/crossbario/autobahn-js/blob/master/test/test_pubsub_multiple_matching_subs.txt

**Comment by oberstet** [comment]
the fix is in PR https://github.com/crossbario/crossbar/pull/1042

**Comment by kuema** [comment]
I tried your test case with the Crossbar versions I tested previously. All tests run fine on CB `16.10.1` and `17.2.1`. It "correctly" stalls on `17.3.1` after receiving 4 messages, waiting for the remaining messages, which would not arrive using this version. However, I had to kill it manually since there does not seem to be a timeout for the test to fail.

Good news is, your PR #1042 resolved the issue. All test cases run fine again. Great work and thank you!
